### PR TITLE
[blockly] Document all new changes for 4.0

### PR DESCRIPTION
**Pull request by stefan-hoehn**
This comprises the documentation for all Blockly related to 4.0

related PRs are:

- https://github.com/openhab/openhab-webui/pull/1617
- https://github.com/openhab/openhab-webui/pull/1722
- https://github.com/openhab/openhab-webui/pull/1664
- https://github.com/openhab/openhab-webui/pull/1723
- https://github.com/openhab/openhab-webui/issues/1309

### CODE REVIEW

File: configuration/blockly/rules-blockly-before-using.md [hunk 1]
```diff
@@ -93,6 +93,28 @@ It should also be mentioned that each of the blocks do have a context sensitive
 
 ![blockly-context.png](../images/blockly/blockly-contextmenu.png)
 
+- **Cross-Rule-Copy/Paste**
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-before-using.md:96]:
```suggestion
#### **Cross-Rule-Copy/Paste**
```

File: configuration/blockly/rules-blockly-items-things.md [hunk 2]
```diff
@@ -195,9 +193,64 @@ More about that topic can be viewed at ![youtube](../images/blockly/youtube-logo
...
+Note: Metadata blocks are not available for Nashorn but only for GraalJS / JS Scripting.
+
+Here is an _expire_ metadata definition which is visualized as
+
+![metadata-expire](../images/blockly/blockly-metadata-expire.png)
+
+and modelled in YAML as
+
+````yaml
+value: 0h3m0s
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-items-things.md:213]:
```suggestion

```yaml
value: 0h3m0s
```

File: configuration/blockly/rules-blockly-items-things.md [hunk 3]
```diff
@@ -195,9 +193,64 @@ More about that topic can be viewed at ![youtube](../images/blockly/youtube-logo
...
+config:
+  ignoreStateUpdates: "true"
+  ignoreCommands: "true"
+````
+
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-items-things.md:218]:
```suggestion
  ignoreCommands: "true"
```

```

File: configuration/blockly/rules-blockly-items-things.md [hunk 4]
```diff
@@ -195,9 +193,64 @@ More about that topic can be viewed at ![youtube](../images/blockly/youtube-logo
...
+    two: "456"
+  one:
+    two:
+      three: "756"
+  field1: 1
+  field2: 2
+  field3: 3
+```
+
+which allows deep hierarchy property access via a configkey-dot-notation like one.two.three
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-items-things.md:246]:
```suggestion
which allows deep hierarchy property access via a configkey-dot-notation like `one.two.three`
```

File: configuration/blockly/rules-blockly-standard-ext.md [hunk 5]
```diff
@@ -19,6 +19,31 @@ This section explains only the blocks that have been added to the standard block
 One of the most commonly used standard blocks are conditions blocks.
 More about conditions can be viewed at ![youtube](../images/blockly/youtube-logo-small.png) [How to use IF and ELSE](https://youtu.be/hSRfooBKn9A?t=445).
 
+## Math
+
+The Math section mainly consists of standard Blockly blocks.
+
+two new math blocks were added for bitwise operations:
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-standard-ext.md:26]:
```suggestion
Two new math blocks were added for bitwise operations:
```

File: configuration/blockly/rules-blockly-uom.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+The following example block gives a good idea of what can be done with the Unit of Measurement blocks
+
+![blockly-uom-example](../images/blockly/blockly-uom-example.png)
+
+[[toc]]
+
+## Overview of the Items and Things category
+
+> ![blockly-uom](../images/blockly/blockly-uom.png)
```
Comment (Confectrician) [inline configuration/blockly/rules-blockly-uom.md:26]:
```suggestion

![blockly-uom](../images/blockly/blockly-uom.png)
```

**Review by Confectrician — COMMENTED** [review]
Just some small changes.

**Review by Confectrician — APPROVED** [review]
Thanks. 👍🏼

**Review by florian-h05 — COMMENTED** [review]
I‘m too late to the party, but it looks good 👍

**Pull request merged**